src/components/views/rooms/wysiwyg_composer/hooks/useInputEventProcessor.ts

```diff
@@ -14,40 +14,164 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { WysiwygEvent } from "@matrix-org/matrix-wysiwyg";
+import { Wysiwyg, WysiwygEvent } from "@matrix-org/matrix-wysiwyg";
 import { useCallback } from "react";
+import { MatrixClient } from "matrix-js-sdk/src/matrix";
 
 import { useSettingValue } from "../../../../../hooks/useSettings";
+import { getKeyBindingsManager } from "../../../../../KeyBindingsManager";
+import { KeyBindingAction } from "../../../../../accessibility/KeyboardShortcuts";
+import { findEditableEvent } from "../../../../../utils/EventUtils";
+import dis from "../../../../../dispatcher/dispatcher";
+import { Action } from "../../../../../dispatcher/actions";
+import { useRoomContext } from "../../../../../contexts/RoomContext";
+import { IRoomState } from "../../../../structures/RoomView";
+import { ComposerContextState, useComposerContext } from "../ComposerContext";
+import EditorStateTransfer from "../../../../../utils/EditorStateTransfer";
+import { useMatrixClientContext } from "../../../../../contexts/MatrixClientContext";
+import { isCaretAtEnd, isCaretAtStart } from "../utils/selection";
+import { getEventsFromEditorStateTransfer } from "../utils/event";
+import { endEditing } from "../utils/editing";
 
-function isEnterPressed(event: KeyboardEvent): boolean {
-    // Ugly but here we need to send the message only if Enter is pressed
-    // And we need to stop the event propagation on enter to avoid the composer to grow
-    return event.key === "Enter" && !event.shiftKey && !event.ctrlKey && !event.metaKey && !event.altKey;
-}
-
-export function useInputEventProcessor(onSend: () => void): (event: WysiwygEvent) => WysiwygEvent | null {
+export function useInputEventProcessor(
+    onSend: () => void,
+    initialContent?: string,
+): (event: WysiwygEvent, composer: Wysiwyg, editor: HTMLElement) => WysiwygEvent | null {
+    const roomContext = useRoomContext();
+    const composerContext = useComposerContext();
+    const mxClient = useMatrixClientContext();
     const isCtrlEnter = useSettingValue<boolean>("MessageComposerInput.ctrlEnterToSend");
```

Comment:
Maybe picky, but could we rename this to `isCtrlEnterToSend`? I know it's a bit longer but I think the clarity is helpful, I was reading this as meaning something like "the user has pressed control and enter at the same time" whereas it actually means "the user has got the setting ctrl enter to send turned on".